### PR TITLE
Make lists sortable with ordering setting 

### LINKED ISSUE #2355: List of content items should be sortable.
They can be sorted in the recipe, but not in the admin.
image

### PR DESCRIPTION
**Pull request by deanmarcussen**
This fixes #2355 

list-ordering

Ordering is optional and configured on the last part settings

When enabled list items are sortable, when disabled they remain sorted by created date - descending

list-part-settings

### CODE REVIEW

**Comment by Skrypt** [comment]
I'm confused, I'm trying to find the Vue.js component but it seems it's initialized with JQuery?

**Comment by deanmarcussen** [comment]
> I'm confused, I'm trying to find the Vue.js component but it seems it's initialized with JQuery?

Yes, I looked at the AdminMenu as you suggested, but that was done with jQuery nested sortable.
I did consider it being a Vue component, but did it this way as it meant all the js and markup could be left out of the page if ordering was disabled (as it is by default).

Figured a vue component would really suit for the other part of your notes on https://github.com/OrchardCMS/OrchardCore/issues/2355 regarding adding existing items and detaching from https://github.com/OrchardCMS/OrchardCore/issues/3667.

Happy to reconsider if you prefer it being done with Vue?

**Comment by Skrypt** [comment]
If the feature you are adding works it's fine merging it. I'm just surprised to see it's using JQuery instead of a full Vue.js component, but like you say it can be added later on. Up to you to do the changes on this PR or on an other one later on.

**Comment by deanmarcussen** [comment]
Probably another PR for the other features - not totally sure how add / move / detach items would work / look like yet

**Comment by deanmarcussen** [comment]
@Skrypt @sebastienros Fixed up some merge conflicts on this old PR.

Is it wanted?

**Comment by Skrypt** [comment]
Sure it is 😄 
I'm just busy 😢

**Comment by agriffard** [comment]
Conflicts due to Docs and Fielsets PRs that have been merged recently. 
Please solve them @deanmarcussen

**Comment by agriffard** [comment]
Please solve conflicts @deanmarcussen

**Comment by deanmarcussen** [comment]
@agriffard conflicts resolved. tested as it's so old this pr, but still works fine!

**Comment by Skrypt** [comment]
Ok reviewing this one now.

**Comment by Skrypt** [comment]
It's still using netstandard 2.0 so we can't merge this.

**Comment by deanmarcussen** [comment]
> It's still using netstandard 2.0 so we can't merge this.

@Skrypt how do you figure that?

Here's the complete `.csproj` from that branch. On core3 as far as I can see... (note did just re sort project references)
```
<Project Sdk="Microsoft.NET.Sdk.Razor">

  <PropertyGroup>
    <TargetFramework>$(AspNetCoreTargetFramework)</TargetFramework>
    <AddRazorSupportForMvc>true</AddRazorSupportForMvc>
  </PropertyGroup>

  <ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Admin.Abstractions\OrchardCore.Admin.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.AdminMenu.Abstractions\OrchardCore.AdminMenu.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentLocalization.Abstractions\OrchardCore.ContentLocalization.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentManagement.Abstractions\OrchardCore.ContentManagement.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentManagement.Display\OrchardCore.ContentManagement.Display.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentTypes.Abstractions\OrchardCore.ContentTypes.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Media.Abstractions\OrchardCore.Media.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.MetaWeblog.Abstractions\OrchardCore.MetaWeblog.Abstractions.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Module.Targets\OrchardCore.Module.Targets.csproj" />
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Users.Abstractions\OrchardCore.Users.Abstractions.csproj" />   
    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ResourceManagement.Abstractions\OrchardCore.ResourceManagement.Abstractions.csproj" />
    <ProjectReference Include="..\OrchardCore.Contents\OrchardCore.Contents.csproj" />
  </ItemGroup>

</Project>
```

**Comment by Skrypt** [comment]
I think I know what I did wrong then

**Comment by Skrypt** [comment]
I made some small css fixes but I can't push to your fork.

**Comment by Skrypt** [comment]
Ok it worked. The commits are pushed. I think it can be merged now.

**Comment by deanmarcussen** [comment]
cheers @Skrypt changes LGTM. @agriffard if you feel like merging :)

**Comment by agriffard** [comment]
@deanmarcussen What about a démo in today's meeting?

**Comment by deanmarcussen** [comment]
Why not indeed Antoine :)

File: src/OrchardCore.Modules/OrchardCore.Lists/Drivers/ListPartDisplayDriver.cs [hunk 1]
```diff
@@ -72,29 +59,22 @@ public override IDisplayResult Display(ListPart listPart, BuildPartDisplayContex
                 );
         }
 
-        private async Task<PagerSlim> GetPagerAsync(IUpdateModel updater, ListPart part)
+        private async Task<PagerSlim> GetPagerAsync(BuildPartDisplayContext context)
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lists/Drivers/ListPartDisplayDriver.cs]: GetPagerSlimAsync

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Lists/Drivers/ListPartDisplayDriver.cs]: > It should be GetPagerSlimAsync and we should also have GetPagerAsync

Changed to `GetPagerSlimAsync`. 
What did you mean by also having `GetPagerAsync`? As a settings option too have the other type of pager?

**Pull request merged**